packages/react-composites/src/composites/MeetingComposite/ChatButtonLogicWrapper.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ChatMessage } from '@azure/communication-chat';
+import React, { useState } from 'react';
+import { useEffect } from 'react';
+import { ChatAdapter } from '../ChatComposite';
+import { ChatButton, ChatButtonProps } from './ChatButton';
+
+/**
+ * @private
+ */
+export type ChatButtonLogicWrapperProps = {
+  chatButtonProps: ChatButtonProps;
+  chatAdapter: ChatAdapter;
+  showChat: boolean;
+  onChatButtonClicked: () => void;
+  disabled: boolean;
+};
+
+/**
+ * @private
+ */
+export const ChatButtonLogicWrapper = (props: ChatButtonLogicWrapperProps): JSX.Element => {
+  const { chatButtonProps, chatAdapter, showChat, onChatButtonClicked, disabled } = props;
+  const [unreadChatMessagesCount, setUnreadChatMessagesCount] = useState<number>(0);
+  useEffect(() => {
+    if (showChat === true) {
+      setUnreadChatMessagesCount(0);
+      return;
+    }
+    const incrementUnreadtChatMessagesCount = (event: { message: ChatMessage }): void => {
+      if (!showChat && event.message.senderDisplayName !== '') {
```

Comment:
can you pull the message checks into a helper function
```
const validNewChatMessage(message) => !!message.senderDisplayName && (message.type === 'text' || message.type === 'html')

Reply:
yup! will do!